### PR TITLE
BI-5544: empty connection error

### PR DESCRIPTION
**Pull request by br0vk1n-ia**

### CODE REVIEW

File: lib/dl_api_lib/dl_api_lib/exc.py [hunk 1]
```diff
@@ -24,6 +24,11 @@ class UnsupportedForEntityType(DLBaseException):
     default_message = "This entity type does not support this operation"
 
 
+class BadConnectionType(DLBaseException):
+    err_code = DLBaseException.err_code + ["BAD_CONN_TYPE"]
+    default_message = "Invalid connection type's value"
```
Comment (MCPN) [inline lib/dl_api_lib/dl_api_lib/exc.py:29]: Invalid connection type value

File: lib/dl_api_lib/dl_api_lib/app/control_api/resources/connections.py [hunk 2]
```diff
@@ -128,9 +128,12 @@ def post(self):  # type: ignore  # TODO: fix
         us_manager = self.get_us_manager()
 
         conn_availability = self.get_service_registry().get_connector_availability()
+        if not (request.json and request.json.get("type") in ConnectionType):
+            raise exc.BadConnectionType()
```
[the remark this answers is not in the document]
Comment (br0vk1n-ia) [reply lib/dl_api_lib/dl_api_lib/app/control_api/resources/connections.py]: fixed

File: lib/dl_api_lib/dl_api_lib/app/control_api/resources/connections.py [hunk 3]
```diff
@@ -128,9 +128,11 @@ def post(self):  # type: ignore  # TODO: fix
         us_manager = self.get_us_manager()
 
         conn_availability = self.get_service_registry().get_connector_availability()
-        conn_type_is_available = conn_availability.check_connector_is_available(
-            ConnectionType[request.json.get("type")]  # type: ignore  # 2024-01-24 # TODO: Item "None" of "Any | None" has no attribute "get"  [union-attr]
-        )
+        conn_type = request.json and request.json.get("type")
+        if not (conn_type and conn_type in ConnectionType):
```
Comment (MCPN) [inline lib/dl_api_lib/dl_api_lib/app/control_api/resources/connections.py:132]: better to expand brackets (`not conn_type or conn_type not in ConnectionType`) for readability

Comment (br0vk1n-ia) [reply lib/dl_api_lib/dl_api_lib/app/control_api/resources/connections.py]: fixed

**Review by MCPN — APPROVED** [review]

**Pull request merged**